extension.json

```diff
@@ -157,6 +157,59 @@
       },
       "libPath": "dist/lib/actions/setOptInPreferences/index.js",
       "viewPath": "actions/setOptInPreferences.html"
+    },
+    {
+      "displayName": "Set Customer IDs",
+      "name": "set-customer-ids",
+      "schema": {
+        "$schema": "http://json-schema.org/draft-04/schema#",
+        "type": "object",
+        "properties": {
+          "instanceName": {
+            "type": "string",
+            "minLength": 1
+          },
+          "customerIds": {
+            "type": "array",
+            "items": {
+              "type": "object",
+              "properties": {
+                "namespace": {
+                  "type": "string",
+                  "minLength": 1
+                },
+                "id": {
+                  "type": "string",
+                  "minLength": 1
+                },
+                "authenticatedState": {
+                  "type": "string",
+                  "minLength": 1
```

Comment:
I'd use an enumeration here to force `authenticatedState` to be one of your three valid values.